### PR TITLE
Add Hatexplain Dataset

### PR DESCRIPTION
**Pull request by kushal2000**
Adding Hatexplain - the first benchmark hate speech dataset covering multiple aspects of the issue

### CODE REVIEW

File: datasets/hatexplain/README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    "post_tokens": ["and","this","is","why","i","end","up","with","nigger","trainee","doctors","who","can","not","speak","properly","lack","basic","knowledge","of","biology","it","truly","scary","if","the","public","only","knew"]
+  }
+}
+~~~
+
+### Data Fields
+
+:small_blue_diamond:post_id : Unique id for each post<br/>
+:small_blue_diamond:annotators : The list of annotations from each annotator<br/>
+:small_blue_diamond:annotators[label] : The label assigned by the annotator to this post. Possible values: [Hatespeech, Offensive, Normal]<br/>
```
Comment (lhoestq) [inline datasets/hatexplain/README.md:107]:
```suggestion
:small_blue_diamond:annotators[label] : The label assigned by the annotator to this post. Possible values: `hatespeech` (0), `normal` (1) or `offensive` (2)<br/>
```
we can make the correspondence with the label ids explicit here

File: datasets/hatexplain/README.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    ],
+    "rationales":[
+    [0,0,0,0,0,0,0,0,1,0,0,1,1,1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,0],
+    [0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],
+    [0,0,0,0,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
+    ],
+    "post_tokens": ["and","this","is","why","i","end","up","with","nigger","trainee","doctors","who","can","not","speak","properly","lack","basic","knowledge","of","biology","it","truly","scary","if","the","public","only","knew"]
+  }
+}
+~~~
```
Comment (lhoestq) [inline datasets/hatexplain/README.md:101]: Update the format to match the one that is actually returned by the dataset object

````suggestion
```
{
    "id": "24198545_gab",
    "annotators": [
      {
        "label": 0,  # hatespeech
        "annotator_id": 4,
        "target": ["African"]
      },
      {
        "label": 0,  # hatespeech
        "annotator_id": 3,
        "target": ["African"]
      },
      {
        "label": 2,  # offensive
        "annotator_id": 5,
        "target": ["African"]
      }
    ],
    "rationales":[
    [0,0,0,0,0,0,0,0,1,0,0,1,1,1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,0],
    [0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],
    [0,0,0,0,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
    ],
    "post_tokens": ["and","this","is","why","i","end","up","with","nigger","trainee","doctors","who","can","not","speak","properly","lack","basic","knowledge","of","biology","it","truly","scary","if","the","public","only","knew"]
  }
}
```
````

File: datasets/hatexplain/README.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+[Needs More Information]
+
+### Licensing Information
+
+[Needs More Information]
+
+### Citation Information
+
+[Needs More Information]
```
Comment (lhoestq) [inline datasets/hatexplain/README.md:175]:
````suggestion
```bibtex
@misc{mathew2020hatexplain,
      title={HateXplain: A Benchmark Dataset for Explainable Hate Speech Detection}, 
      author={Binny Mathew and Punyajoy Saha and Seid Muhie Yimam and Chris Biemann and Pawan Goyal and Animesh Mukherjee},
      year={2020},
      eprint={2012.10289},
      archivePrefix={arXiv},
      primaryClass={cs.CL}
}
```
````

File: datasets/hatexplain/hatexplain.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+import csv
+import json
+import os
+
+import datasets
+
+
+_CITATION = """\
+@misc{mathew2020hatexplain,
+      title={HateXplain: A Benchmark Dataset for Explainable Hate Speech Detection},
```
Comment (lhoestq) [inline datasets/hatexplain/hatexplain.py:28]:
```suggestion
      title={HateXplain: A Benchmark Dataset for Explainable Hate Speech Detection},
```
trailing whitespace

File: datasets/hatexplain/hatexplain.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Hatexplain: A Benchmark Dataset for Explainable Hate Speech Detection"""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import json
+import os
```
Comment (lhoestq) [inline datasets/hatexplain/hatexplain.py:21]:
```suggestion
import json
```
unused imports

**Review by lhoestq — COMMENTED** [review]
Nice thank you !

I added suggestions that may fix the CI

File: datasets/hatexplain/README.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+- **Homepage:** [Needs More Information]
+- **Repository:** https://github.com/punyajoy/HateXplain/
+- **Paper:** https://arxiv.org/abs/2012.10289
+- **Leaderboard:** [Needs More Information]
+- **Point of Contact:** [Needs More Information]
+
+### Dataset Summary
+
+Hatexplain is the first benchmark hate speech dataset covering multiple aspects of the issue. Each post in the dataset is annotated from three different perspectives: the basic, commonly used 3-class classification (i.e., hate, offensive or normal), the target community (i.e., the community that has been the victim of hate speech/offensive speech in the post), and the rationales, i.e., the portions of the post on which their labeling decision (as hate, offensive or normal) is based.
```
Comment (lhoestq) [inline datasets/hatexplain/README.md:57]:
```suggestion
Hatexplain is the first benchmark hate speech dataset covering multiple aspects of the issue. Each post in the dataset is annotated from three different perspectives: the basic, commonly used 3-class classification (i.e., hate, offensive or normal), the target community (i.e., the community that has been the victim of hate speech/offensive speech in the post), and the rationales, i.e., the portions of the post on which their labeling decision (as hate, offensive or normal) is based.

WARNING: This dataset contains content that are offensive and/or hateful in nature.
```

We can also make it explicit that the content can be offensive and hateful

File: datasets/hatexplain/README.md (hunk 3 above)
Comment (kushal2000) [reply datasets/hatexplain/README.md]: Outdated suggestion. I will make a fresh commit

File: datasets/hatexplain/README.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+multilinguality:
+- monolingual
+size_categories:
+- 10K<n<100K
+source_datasets:
+- original
+task_categories:
+- text-classification
+task_ids:
+- sentiment-classification
```
Comment (lhoestq) [inline datasets/hatexplain/README.md:19]:
```suggestion
- text-classification-other-hate-speech-detection
```

**Review by lhoestq — APPROVED** [review]
Thanks :)

**Pull request merged**